chain/db/db.go

```diff
@@ -33,6 +42,13 @@ func NewDB(dialect, path string, metadata map[string]string) (*BandDB, error) {
 		"RESTRICT",
 	)
 
+	db.Model(&DataSourceRevision{}).AddForeignKey(
+		"data_source_id",
+		"data_sources(id)",
+		"CASCADE",
```

Comment:
Why don't we do RESTRICT? I think it's safer that way.

Reply:
In my opinion, this table will never be deleted by logic, but it's okay to use RESTRICT